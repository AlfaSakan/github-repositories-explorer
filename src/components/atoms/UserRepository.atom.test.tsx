import { describe, it, expect, vi } from "vitest";
import { fireEvent, render, screen } from "@testing-library/react";
import UserRepository from "./UserRepository.atom";

describe("UserRepository", () => {
  it("should match the snapshot", () => {
    const { container } = render(
      <UserRepository
        user={{
          id: 1,
          login: "login",
          repos_url: "",
          repositories: [
            {
              description: "description",
              id: 2,
              name: "name",
              stargazers_count: 111,
            },
          ],
          show: true,
        }}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("should invoke click function", async () => {
    const mockFn = vi.fn();
    render(
      <UserRepository
        onToggle={mockFn}
        user={{
          id: 1,
          login: "login 1",
          repos_url: "",
          repositories: [
            {
              description: "description",
              id: 2,
              name: "name",
              stargazers_count: 111,
            },
          ],
          show: true,
        }}
      />
    );
    fireEvent.click(screen.getByText("login 1"));
    expect(mockFn).toHaveBeenCalled();
  });
});
