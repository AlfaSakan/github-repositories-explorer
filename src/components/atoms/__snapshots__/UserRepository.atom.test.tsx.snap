// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserRepository > should match the snapshot 1`] = `
<div>
  <div
    class="mb-3"
  >
    <button
      class="h-10 flex px-3 items-center w-full justify-between bg-neutral-100"
    >
      <p>
        login
      </p>
      <svg
        class="lucide lucide-chevron-down"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m6 9 6 6 6-6"
        />
      </svg>
    </button>
    <div
      class="space-y-3 mt-3 ml-3"
    >
      <div
        class="bg-neutral-400 px-2 py-4"
      >
        <div
          class="flex items-center justify-between"
        >
          <p
            class="font-semibold"
          >
            name
          </p>
          <div
            class="flex items-center"
          >
            111
            <svg
              class="lucide lucide-star inline fill-black"
              fill="none"
              height="16"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
              />
            </svg>
          </div>
        </div>
        <p>
          description
        </p>
      </div>
    </div>
  </div>
</div>
`;
